### PR TITLE
Integrate mini draft petition generation

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-585

### Change description ###

- Includes new field to capture divorce or dissolution for solicitor create petition journey.
- This PR wires in functionality to generate mini draft petition which is generated using docmosis template and document is stored in DM store. Doc assembly service is used which generates the pdf using docmosis template and uploads to dm store.
- This is tested end to end on local docker environment.

- Functional test will be created as a separate PR.
- There is one minor change required in docmosis template(removing extra for) which will done in docmosis repository.

image

image (25)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorCreateTest.java [hunk 1]
```diff
@@ -92,6 +140,7 @@ void givenValidCaseDataWhenAboutToSubmitCallbackIsInvokedCaseDataIsSetCorrectly(
             .getResponse()
             .getContentAsString();
 
+        System.out.println(jsonStringResponse);
```
Comment (linusnorton) [inline src/integrationTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorCreateTest.java:143]: remove please

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/MiniPetitionDraft.java [hunk 2]
```diff
@@ -47,13 +47,13 @@ public CaseDataContext updateCaseData(CaseDataContext caseDataContext, CaseDataU
             languagePreference = WELSH;
         }
 
-        String templateId = docmosisTemplatesConfig.getTemplates().get(languagePreference).get(DIVORCE_MINI_PETITION);
+        String templateName = docmosisTemplatesConfig.getTemplates().get(languagePreference).get(DIVORCE_MINI_PETITION);
 
         DocumentInfo documentInfo = docAssemblyService.generateAndStoreDraftPetition(
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/service/MiniPetitionDraft.java:52]: this method should be called renderDocument

File: src/main/java/uk/gov/hmcts/divorce/document/mapper/CaseDataToDraftPetitionTemplateMapper.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.MARRIAGE_OR_RELATIONSHIP;
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.OF_THE_DIVORCE;
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.RELATIONSHIP;
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.RESPONDENT_POSTAL_ADDRESS;
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.TO_END_A_CIVIL_PARTNERSHIP;
+import static uk.gov.hmcts.divorce.document.mapper.DocmosisTemplateConstants.TO_END_THE_CIVIL_PARTNERSHIP;
+
+@Component
+@Slf4j
+public class CaseDataToDraftPetitionTemplateMapper {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/document/mapper/CaseDataToDraftPetitionTemplateMapper.java:50]: I would call this DraftPetitionTemplateContent and the method apply as it seems to follow the same pattern as CommonContent

**Review by mapingo — APPROVED** [review]

**Comment by nitinprabhu** [comment]
Closing this and creating new one as the pr ccd seems corrupted

**Pull request closed without merging**